docs/lib/graphqlapi/fragments/ios/query-data.md

```diff
@@ -121,3 +120,41 @@ func listTodos() -> AnyCancellable {
 </amplify-block>
 
 </amplify-block-switcher>
+
+> **Note**: This approach will only return up to the first 1,000 items.  To change this limit or make requests for additional results beyond this limit, use *pagination* as discussed below.
+
+## List subsequent pages of items
```

Comment:
This example only shows how to get 2 pages of results.  Can you provide an example of getting all pages?

Reply:
Updated example to contain both `getNextPage()` and `getAllPages()`

Reply:
I think this example is a little confusing now because it is showing two use cases in one code block.  I'd suggest only showing how to retrieve all pages, or splitting this into two independent code blocks.  

On another note, the overall DX does not seem great.  Customers basically have to implement the success and failure callbacks twice - once for fetching the first page, and a second time for fetching subsequent pages.   I doubt there's a use case when the logic would differ for these scenarios.  Has this new API gone through an API review yet?

Reply:
thanks for the feedback, i've updated the sample and separated them into two. one that shows how to load the next page and another that retrieves all pages at once

in regards to the DX, I was working around a few things in mind such as 
- how to support pagination from the list instance for both DataStore and API users. Developers can create a List object from Amplify.DataStore APIs and Amplify.API APIs. 
- thus, how we can share the error handling, which introduced a new Amplify CoreError type that wraps the APIError or DataStore error. This is probably the reason why the DX seems different across the two calls, ie. Amplify.API returns two layers of results (which i think is a painpoint that we want to revisit) and `getNextPage` returns a `Result<List, CoreError>`. 
I'd be happy to grab some of your time to iterate on the DX, and I've gone through a couple of design/implementation reviews with tim and daniel on this. I believe the lack of cross platform reviews was a conscious decision on iOS side to move forward with these APIs to have feature parity with Android as we recognized that the solution was becoming very iOS platform dependent (if you recall, i actually wrote some code that `getNextPage` returns a `GraphQLRequest` and at that point we met to discuss the need to iterate on the solution). I expect these APIs to change or influence the overall cross platform efforts for introducing lazy loading in the future (loading associated data from the list instance) since they are all methods part of the List type